Assets/Scripts/GameManager.cs

```diff
@@ -56,67 +56,84 @@ public void StartGame(PlayerSupervisor supervisor)
         supervisor.StartGame();
     }
 
-    public void WinGame(PlayerSupervisor supervisor)
+    public void PlayerLostBall(PlayerSupervisor supervisor)
     {
         if (trainingMode)
         {
+            supervisor.LoseGame();

```

Comment:
```suggestion
    public void PlayerLostBall(PlayerSupervisor supervisor)
    {
        supervisor.LoseGame();
        if (trainingMode)
        {
```
Having GameManager call LoseGame each time the player loses a ball seems to fix the problem I mentioned in my comment on PlayerSupervisor.cs. This might be fine though semantically I could see how calling LoseGame before a game mode ends is confusing. An alternative would be just making a small function for PlayerSupervisor to reset its playerState back to Waiting. I'll defer to your judgement on that.